src/main/java/com/ticketbooking/service/ReservationService.java

```diff
@@ -0,0 +1,323 @@
+
+        screeningDto.setRoomNumber(room.getRoomNumber());
+        List<Seat> seatAllList = seatRepo.findAllByRoomId(room.getId());
+
+        List<SeatDto> seatDtos = seatAllList.stream()
+                .map(s -> {
+                    SeatDto seatDto = new SeatDto
+                            (s.getId(), s.getRowNr(), s.getSeatNr(), s.getRoom().getId());
+                    return seatDto;
+                })
+                .collect(Collectors.toList());
+
+        screeningDto.setSeats(seatDtos);
+
+        return screeningDto;
+    }
+
+    private Supplier<TicketBookingNotFoundException> getNotFoundExceptionSupplier(String key, String[] values) {
+        return () -> new TicketBookingNotFoundException(getMessage(key, values));
+    }
+
+    private String getMessage(String key, String[] values) {
+        return messageSource.getMessage (key, values, LocaleContextHolder.getLocale());
+    }
+
+    @Transactional
+    public String makeReservation(ReservationDto reservationDto, HttpServletRequest request) {
+        StringBuilder response = new StringBuilder();
+        Screening screening = screeningRepo.findById(reservationDto.getScreeningId())
+            .orElseThrow(getNotFoundExceptionSupplier("screening.not.found", new String[] {String.valueOf(reservationDto.getScreeningId())}));
+
+        if (LocalDateTime.now().plusMinutes(15).isAfter(screening.getScreeningTime())) {
+            response.append(getMessage("booking.time.expired", null));
+        } else if (reservationDto.getTickets().isEmpty()) {
+            response.append(getMessage("one.place.reservation.rule", null));
+        } else if (!checkClientName(reservationDto)) {
+            response.append(getMessage("name.reservation.rule", null));
+        } else if (!checkClientSurname(reservationDto)) {
+            response.append(getMessage("surname.reservation.rule", null));
+        } else if (!checkSingleSeat(reservationDto, screening)) {
+            response.append(getMessage("new.place.reservation.rule", null));
+        } else {
+            Reservation reservation = saveReservation(reservationDto, screening);
+            BigDecimal ticketsPrice = saveTickets(reservationDto, screening, reservation);
+
+            sendReservationByMail(request, reservation.getId(), reservationDto.getMail());
+            
+            response.append(getMessage("confirm.reservation.line.one", null));
+            response.append(getMessage("confirm.reservation.line.two", new String[] {reservationDto.getName(), reservationDto.getSurname()}));
+            response.append(getMessage("confirm.reservation.line.three", new String[] {ticketsPrice.toString()}));
+            response.append(getMessage("confirm.reservation.line.four", null));
+        }
+
+        return response.toString();
+    }
+
+    private boolean checkClientSurname(ReservationDto reservationDto) {
```

Comment:
Czemu na siłę zmieniamy te pierwsze literki? Poza tym to sprawdzasz tylko długość, ktoś może wpisać `472389FJF934!@$` i przejdzie, prawda?